lib/ufe/UsdStageMap.cpp

```diff
@@ -31,32 +68,52 @@ UsdStageMap g_StageMap;
 
 void UsdStageMap::addItem(const Ufe::Path& path, UsdStageWeakPtr stage)
 {
-	fPathToStage[path] = stage;
-	fStageToPath[stage] = path;
+    // We expect a path to the proxy shape node, therefore a single segment.
+    auto nbSegments = path.nbSegments();
+    if (nbSegments != 1) {
+        TF_CODING_ERROR("A proxy shape node path can have only one segment, path '%s' has %d", path.string().c_str(), nbSegments);
+        return;
+    }
+            
+    // Convert the tail of the UFE path to an MObjectHandle.
+    auto proxyShape = proxyShapeHandle(path);
+    if (!proxyShape.isValid()) {
+        return;
+    }
+    
+    // Could get the stage from the proxy shape object in the stage() method,
+    // but since it's given here, simply store it.
+    fObjectToStage[proxyShape] = stage;
+    fStageToObject[stage] = proxyShape;
 }
 
 UsdStageWeakPtr UsdStageMap::stage(const Ufe::Path& path) const
 {
+    auto proxyShape = proxyShapeHandle(path);
+    if (!proxyShape.isValid()) {
+        return nullptr;
+    }
```

Comment:
Looks like maybe a leading space missing from this block to align it with the code below.

Reply:
Tabified in
https://github.com/Autodesk/maya-usd/pull/237/commits/f53f29c7258bc1d99897d843dbfef2e0316dd702